db-engine/src/main/java/com/splicemachine/utils/DbEngineUtils.java

```diff
@@ -0,0 +1,17 @@
+package com.splicemachine.utils;
+
+public class DbEngineUtils {
+
+    public static String getJavaRegexpFilterFromAsterixFilter(String asterixFilter)
+    {
+        String filter = asterixFilter;
+        String toEscape[] = {"<", "(", "[", "{", "^", "-", "=", "$", "!", "|", "]", "}", ")", "+", ".", ">"};
+        for(String s : toEscape) {
+            filter = filter.replaceAll("\\" + s, "\\\\" + s);
+        }
+        filter.replaceAll("\\\\", "\\\\\\\\");
```

Comment:
maybe a small unit test for this case, why we need to escape the escape.

Reply:
done

Reply:
now really done :-)